### ISSUE TITLE
KeyError: 'also_return_response' exception with bravado-core 4.10.0

### ISSUE THREAD

**Issue by elmirjagudin** [Yelp/bravado#326]
The hello word example from documentation fails with KeyError: 'also_return_response'
exception. This is with bravado 9.1.1 and bravado-core 4.10.0 packages installed.

```
>>> from bravado.client import SwaggerClient
>>> client = SwaggerClient.from_url('http://petstore.swagger.io/v2/swagger.json')
<....>/lib/python3.5/site-packages/bravado_core/spec.py:144: Warning: config also_return_response is been removed because is not a recognized config key
  category=Warning,
>>> pet = client.pet.getPetById(petId=42).result()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<....>/lib/python3.5/site-packages/bravado/client.py", line 261, in __call__
    config['also_return_response'])
KeyError: 'also_return_response'
```

A work-around is to downgrade bravado-core to 4.9.1, then at least hello world example works.

I'm guessing this is related to this change in bravado-core:
https://github.com/Yelp/bravado-core/commit/aecd9122f5301973f0114ca3f101657ce6b9e046

Also these Issue and PR seems related:
https://github.com/Yelp/bravado-core/issues/217
https://github.com/Yelp/bravado-core/pull/218

**Comment by macisamuele** [comment]
@elmirjagudin thanks for reporting it.
As you noticed we're already aware of this issue and tomorrow with @sjaensch we're going to release a new version of ``bravado`` or ``bravado-core`` that fixes the issue.

As temporary workaround you can use an "older" version of ``bravado-core`` (ie. 4.9.1)

Related PRs:
 * [``bravado-core`` #218](https://github.com/Yelp/bravado-core/pull/218)
 * [``bravado`` #325](https://github.com/Yelp/bravado/pull/325)

**Comment by elmirjagudin** [comment]
Cool Samuele! I noticed that you have pushed a fix the minute I submitted the issue report :)

I have tried your https://github.com/Yelp/bravado/pull/325, it solves the bug. Thanks!

**Comment by advance512** [comment]
This broke our system as well.

Pretty ironic that we hard-set the version of bravado to:
`bravado==8.4.0`
to prevent these "3rd party breaks". Yet we still get them as bravado itself depends on:
`bravado-core >= 4.2.2`

bravado should point to specific versions of bravado-core and pyyaml.

**Comment by macisamuele** [comment]
I don't see the point of strictly pinning ``bravado`` dependencies to specified versions.
Doing so will make harder/impossible to use other libraries dependent on a different version of ``pyyaml``.

> Pretty ironic that we hard-set the version of bravado to:
bravado==8.4.0

If you strictly pin ``bravado`` to a specific version you're not constraining the dependencies injected by ``bravado`` to specific versions and for extension the dependencies of the bravado dependencies. Solution to this is by using requirements file (``pip freeze > requirements.txt``)

I would suggest reading [Install requires vs Requirements](https://packaging.python.org/discussions/install-requires-vs-requirements/).
The main idea is:
 * ``setup.py`` describes the list of all direct dependencies and eventually minimum versions due to used features
 * ``requirements.txt`` contains the exact version of all the packages needed on your python environment (including derived dependencies)

The usage of ``requirements.txt`` file allows you to have a reproducible environment.

**Comment by advance512** [comment]
@macisamuele 

Good point. I did indeed consider doing `pip freeze > requirements.txt`, but that would make it harder to follow what libraries we use directly - not the entire set of dependencies (and their dependencies, etc) all around. That way, when we need/want to upgrade these libraries, it is clear which ones we should look for "breaking changes" in.

I am sure there are better solutions, e.g. [Pipfile.lock](https://github.com/pypa/pipfile), which we'll consider using.

For now, seems like bravado-core broke backwards compatibility and should fix that asap.

**Comment by macisamuele** [comment]
In Yelp we're using [requirements-tools](https://github.com/Yelp/requirements-tools) that allows to write a ``requirements-minimal.txt`` file containing you're direct dependencies (more or less like in setup.py), but even pipfile seems a valid option.

About the bravado-core change we have two PRs on the fly to fix it 😄

**Comment by advance512** [comment]
Thanks, appreciated. :)

**Comment by macisamuele** [comment]
[``bravado-core`` #218](https://github.com/Yelp/bravado-core/pull/218) has been merged and released on v4.10.1.
